vue-client/src/components/mixins/sidesearch-mixin.vue

```diff
@@ -136,16 +127,50 @@ export default {
       this.fetch(n);
     },
     getItems(keyword) {
-      let params = this.getSearchParams(this.getSearchPhrase(keyword));
-      params = Object.assign(params, {
+      const searchPhrase = this.getSearchPhrase(keyword);
+      const urlSearchParams = new URLSearchParams({
+        q: searchPhrase,
         _limit: this.maxResults,
         _offset: this.currentPage * this.maxResults,
         _sort: this.sort,
+        ...(typeof this.typeArray !== 'undefined' && this.typeArray.length > 0 && { '@type': this.typeArray }),
+        ...this.currentSearchParam?.mappings,
+        ...this.currentSearchParam?.searchProps.reduce((acc, searchProp) => ({
+          ...acc,
+          [searchProp]: searchPhrase,
+        }), {}),
       });
-      if (typeof this.typeArray !== 'undefined' && this.typeArray.length > 0) {
-        params['@type'] = this.typeArray;
+
+      const field = VocabUtil.getTermObject(this.fieldKey, this.resources.vocab, this.resources.context);
+
+      /**
+       * If field is a kbv:predicate (e.g. role) then filter linkable items depending on field and parent type.
+       * */
+      // A VocabUtil.isSubPropertyOf(field.subPropertyOf, 'predicate', ...) would be preferable here.
+      if (field.subPropertyOf.find(subProp => subProp['@id'] === VocabUtil.getTermObject('predicate', this.resources.vocab, this.resources.context)['@id'])) {
+        const statement = VocabUtil.getTermObject(field.domain[0]['@id'], this.resources.vocab, this.resources.context); // e.g. Contribution
+        const statementOf = statement.allowedProperties.find(p => p.domain?.find(d => d['@id'] === statement['@id'])); // e.g. contributionOf
+        const statementOfRangeIncludes = statementOf.rangeIncludes[0]['@id']; // e.g. Endeavour
```

Comment:
I think these lines should handle 
* domain/range vs domainIncludes/rangeIncludes
* empty domain/range
* number of domainIncludes/rangeIncludes > 1 (by aborting is ok)
?

But it is not important now. We can fix that while refactoring lxljs.

Reply:
I'd like to talk/whiteboard this, since i think:
1.  It _has_ to be `domain` (it's the only fact that entails something; "includes" are _only_ suggestions, they don't _mean_ what this is a property of).
2. The same goes applies for `range`, but I'm not sure why we check all allowedProperties here? (We got here via `contribution` and that is why we're using this logic?)
3. Yes, abort on empty sounds correct.

Reply:
> I'd like to talk/whiteboard this, since i think:
> 
>     1. It _has_ to be `domain` (it's the only fact that entails something; "includes" are _only_ suggestions, they don't _mean_ what this is a property of).

:heavy_check_mark: suspected this

> 
>     2. The same goes applies for `range`, but I'm not sure why we check all allowedProperties here? (We got here via `contribution` and that is why we're using this logic?)

I think that was what was currently available in this context.
We should add better vocab preprocessing + lxljs methods to make clearer.

> 
>     3. Yes, abort on empty sounds correct.

:heavy_check_mark:

Reply:
Actually I think the last line needs to handle all cases?
Out ff the `hasStatement` subproperties, `contribution` and `seriesMembership` have `domainIncludes` while `releationship` has `domain`.

Reply:
Like this?:
```js
const subClassesOfRangeIncludes = [
  ...new Set( // spread Set to ensure unique values in array
    [...(statementOf.range || []), ...(statementOf.rangeIncludes || [])].flatMap( // iterate over both range and rangeIncludes
      rangeItem => VocabUtil.getSubClassChain(rangeItem['@id'], this.resources.vocabClasses, this.resources.context), // get subclasses of e.g. Endeavour
    ),
  ),
];
```

Reply:
> Like this?:
> 
> ```js
> const subClassesOfRanges = [
>   ...new Set( // spread Set to ensure unique values in array
>     [...(statementOf.range || []), ...(statementOf.rangeIncludes || [])].flatMap( // iterate over both range and rangeIncludes
>       rangeItem => VocabUtil.getSubClassChain(rangeItem['@id'], this.resources.vocabClasses, this.resources.context), // get subclasses of e.g. Endeavour
>     ),
>   ),
> ];
> ```

☝️Added in the latest commit.

Reply:
We merge but continue discussion